src/waveform/renderers/waveformmarkset.cpp

```diff
@@ -60,15 +66,33 @@ void WaveformMarkSet::setup(const QString& group, const QDomNode& node,
 
             if (controlItemSet.insert(hotCueControlItem).second) {
                 //qDebug() << "WaveformRenderMark::setup - Automatic mark" << hotCueControlItem;
-                m_marks.push_back(m_defaultMark);
-                WaveformMark& mark = m_marks.back();
-                mark.setKeyAndIndex(pHotcue->getKey(), i);
+                m_marks.push_back(QSharedPointer<WaveformMark>(new WaveformMark()));
+                WaveformMark* mark = m_marks.back().data();
+                WaveformMarkProperties defaultProperties = m_pDefaultMark.properties();
```

Comment:
These 3 lines could be replaced by an explicit constructor in WaveformMark with a single iHotCue argument.